src/partial17.lib/partial17/Partial.test.cpp

```diff
@@ -11,16 +12,48 @@ using namespace partial17;
 
 using meta17::type;
 
-TEST(Partial, basic) {
-    auto x = Partial<char, int, float>{}; //
-    auto y = x; // copy constructor
-    x = y; // copy operator
-    x = std::move(y); // move operator
-    auto z = Partial<char, int, float>{std::move(x)}; // move construct
+TEST(Partial, construction) {
+    using P = Partial<char, int, float>;
+    auto p0 = P{};
+    ASSERT_FALSE(p0.has<0>());
+    ASSERT_FALSE(p0.has<1>());
+    ASSERT_FALSE(p0.has<2>());
+    // TODO CK: Should this be possible? - crashes, would suggest a graceful value for production
+    //    EXPECT_EQ(p0.get<0>(), '\0');
+    //    EXPECT_EQ(p0.get<1>(), 0);
+    //    EXPECT_EQ(p0.get<2>(), 0);
+
+    auto p1 = P{'c', 2.3f};
+    auto p2 = p1; // copy constructor
+    auto p3 = P{std::move(p2)}; // move construct
+
+    EXPECT_EQ(p1.get<0>(), 'c');
+    EXPECT_EQ(p1.get<2>(), 2.3f);
+    EXPECT_EQ(p1.countInitialized(), 2);
+    EXPECT_EQ(p1.countAll(), 3);
+    // TODO CK: Consider alignment - does this need to consider the size of the bitset

```

Comment:
`size()` is simply the size of the heap allocated for the partial. We might want to document this.

Reply:
Since `size()` it returns a binary size, I would suggest changing the naming to something in that direction. The *STL* uses `size()` for the number of elements within a container (which is currently retrievend by `countAll()` and `countInitialized()`).

Reply:
Yes, you are correct in your observation. It's best compared to `capacity` on a `std::vector` but the semantics are different. It seems the STL has the wrong names here. So I am quite undecided what is best. We need a wording scheme for these methods, document it and use it on all value types where appropriate.